File: database/models/MedicalRecord/Lifestyle.js
import { Schema, model } from 'mongoose';

const lifestyleSchema = new Schema({
  userId: {
    type: Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  activityType: {
    type: String,
    required: true,
    enum: ['Tập thể dục', 'Ăn uống', 'Giấc ngủ', 'Hút thuốc', 'Uống rượu', 'Khác']
  },
  description: {
    type: String,
    required: true
  },
  frequency: {
    type: String,
    required: true,
    enum: ['Hàng ngày', 'Hàng tuần', 'Hàng tháng', 'Thỉnh thoảng']
  },
  duration: {
    type: String
  },
  startDate: {
    type: Date,
  },
  notes: {
    type: String
  }
}, {
  timestamps: true
});

export default model('Lifestyle', lifestyleSchema); 

File: mainflutter/lib/models/lifestyle.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'lifestyle.freezed.dart';
part 'lifestyle.g.dart';

@freezed
class Lifestyle with _$Lifestyle {
  const factory Lifestyle({
    @JsonKey(name: '_id') required String id,
    required String activityType,
    required String description,
    required String frequency,
    String? duration,
    @JsonKey(fromJson: _dateTimeFromJson, toJson: _dateTimeToJson) DateTime? startDate,
    String? notes,
  }) = _Lifestyle;

  factory Lifestyle.fromJson(Map<String, dynamic> json) => _$LifestyleFromJson(json);
}

DateTime? _dateTimeFromJson(String? date) => date == null ? null : DateTime.parse(date);

String? _dateTimeToJson(DateTime? date) => date?.toIso8601String(); 